src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -532,77 +502,6 @@ internal Expr BindArrayIndexCore(BindingFlag bindFlags, Expr pOp1, Expr pOp2)
-        {
-            CType typeRet = GetTypes().GetPointer(GetReqPDT(PredefinedType.PT_CHAR));
-
-            return GetExprFactory().CreateUnaryOp(ExpressionKind.EK_ADDR, typeRet, @string);
-        }
-
-        private ExprQuestionMark BindPtrToArray(ExprLocal exprLoc, Expr array)
-        {
-            CType typeElem = array.Type.AsArrayType().GetElementType();
-            CType typePtrElem = GetTypes().GetPointer(typeElem);
-
-            // element must be unmanaged...
-            if (GetSymbolLoader().isManagedType(typeElem))
-            {
-                ErrorContext.Error(ErrorCode.ERR_ManagedAddr, typeElem);
-            }
-
-            SetExternalRef(typeElem);
-
-            Expr test = null;
-            // we need to wrap the array so we can effectively generate something like this:
-            // (((temp = array) != null && temp.Length > 0) ? loc = temp[0] : loc = null)
-            // NOTE: The assignment needs to be inside the ExpressionKind.EK_QUESTIONMARK.
-            // We can't do loc = (... ? ... : ...) since the CLR type of temp[0] is a managed
-            // pointer and null is a UIntPtr - which confuses the JIT. We can't just convert
-            // temp[0] to UIntPtr with a conv.u instruction because then if a GC occurs between
-            // the time of the cast and the assignment to the local, we're toast.
-            ExprWrap wrapArray = WrapShortLivedExpression(array);
-            Expr save = GetExprFactory().CreateSave(wrapArray);
-            Expr nullTest = GetExprFactory().CreateBinop(ExpressionKind.EK_NE, GetReqPDT(PredefinedType.PT_BOOL), save, GetExprFactory().CreateConstant(wrapArray.Type, ConstVal.Get(0)));
-            Expr lenTest;
-
-            if (array.Type.AsArrayType().rank == 1)
-            {
-                Expr len = GetExprFactory().CreateArrayLength(wrapArray);
-                lenTest = GetExprFactory().CreateBinop(ExpressionKind.EK_NE, GetReqPDT(PredefinedType.PT_BOOL), len, GetExprFactory().CreateConstant(GetReqPDT(PredefinedType.PT_INT), ConstVal.Get(0)));
-            }
-            else
-            {
-                ExprCall call = BindPredefMethToArgs(PREDEFMETH.PM_ARRAY_GETLENGTH, wrapArray, null, null, null);
-                lenTest = GetExprFactory().CreateBinop(ExpressionKind.EK_NE, GetReqPDT(PredefinedType.PT_BOOL), call, GetExprFactory().CreateConstant(GetReqPDT(PredefinedType.PT_INT), ConstVal.Get(0)));
-            }
-
-            test = GetExprFactory().CreateBinop(ExpressionKind.EK_LOGAND, GetReqPDT(PredefinedType.PT_BOOL), nullTest, lenTest);
-
-            Expr list = null;
-            Expr pList = list;
-            Expr pLastList = null;
-            for (int cc = 0; cc < array.Type.AsArrayType().rank; cc++)
-            {
-                GetExprFactory().AppendItemToList(GetExprFactory().CreateConstant(GetReqPDT(PredefinedType.PT_INT), ConstVal.Get(0)), ref pList, ref pLastList);
-            }
-            Debug.Assert(list != null);
```

Comment:
There was a bug here, since `list` is never assigned, but rather `pList` is. Since the code is demonstrably unreachable it doesn't matter, but it may indicate a bug in the code this was ported from too, though I would guess it was the porting that introduced the error.